src/DotNet/Writer/MethodBodyWriterBase.cs

```diff
@@ -158,20 +172,19 @@ public abstract class MethodBodyWriterBase {
 		/// <param name="instr">The instruction</param>
 		protected void WriteOpCode(ref ArrayWriter writer, Instruction instr) {
 			var code = instr.OpCode.Code;
-			if ((ushort)code <= 0xFF)
-				writer.WriteByte((byte)code);
-			else if (((ushort)code >> 8) == 0xFE) {
-				writer.WriteByte((byte)((ushort)code >> 8));
-				writer.WriteByte((byte)code);
-			}
-			else if (code == Code.UNKNOWN1)
-				writer.WriteByte((byte)Code.Nop);
-			else if (code == Code.UNKNOWN2)
-				writer.WriteUInt16((ushort)(((ushort)Code.Nop << 8) | Code.Nop));
-			else {
-				Error("Unknown instruction");
-				writer.WriteByte((byte)Code.Nop);
+			if (instr.OpCode.Size == 2)
```

Comment:
This whole method needs to be reformatted (braces on the wrong line, `else` on the same line as `{`)

I also don't like that the whole method got rewritten. Is it not possible to just add the minimal amount of code here to support the new feature without rewriting the whole method?

Reply:
I just rewrote it like this because it seemed kind of unwieldy to me after adding the case for experimental opcodes, and it occurred to me that it makes more sense to check the size rather than special-casing the opcode ranges.

It's not something I feel particularly strongly about. I can retain the old logic if you prefer.